### ISSUE TITLE
Some questions

### ISSUE THREAD

**Issue by xingchenzhang** [FreeformRobotics/EAEFNet#2]
Hi,

Thank you very much for this wonderful work!

I tried to train the segmentation network (with ResNet 152), however I got Acc (69.37) and mIoU (57.37) on MFNet, which have some differences from your reported results in the paper.

Do you have any ideas why this happened?

I noticed that in your paper, you mentioned the batch size was 5. But in the code, this was set to 3. Also, are you reporting the results of the 100th epoch in the paper? Or did you use any other model selection methods to select the final reported results?

In addition, you mentioned there were 420 pairs for testing. But there are actually only 393 pairs for testing. Not sure if I misunderstood anything here.

Thanks!
Xingchen

**Comment by matrixgame2018** [comment]
Regarding some methods for the experiment accuracy not meeting expectations, I have some ideas.

1. Due to the presence of inaccurate annotations in the dataset and a severe class distribution imbalance, the training results are not stable. If the training results can be replicated, I believe an mIoU between 58 and 59 is reasonable.

2. In addition, there is a technique related to this work. On line 184 in the model/EAEFNet, 

   during training, use:

   ` x = self.conv(self.se(self.conv4(x3_2))),` 

   during testing use:

   `x = self.conv(self.conv4(x3_2))`

   With this operation, the mAcc  score of the model will increase. However, based on our observations during this period, the mAcc metric does not accurately reflect the overall performance of the work. I personally suggest abandoning its use.

3. I recommend using the training dataset provided by RTFNet, with batch_size = 5 and lr = 0.02, to conduct experiments that may achieve the expected results.

4. If possible, you can provide me with the environment you used and the testing_results_file.txt file at [2443434059@qq.com](mailto:2443434059@qq.com). Based on your results, I will further analyze and provide more specific solutions!

Regarding discrepancies between the description in some papers and the code, I have some questions that need clarification.

1. Regarding the code part, during the experiment, we used a batch_size of 5, lr of 0.02, and 100 epochs to achieve the accuracy of 58.9 as described in the paper. The batch size of 3 in the code is a tuning version used for GPUs with 12GB VRAM (GTX1080ti, RTX2080ti). If your GPU has 24GB VRAM, you can try using batch_size=5 for the experiment.
2. The part about *using 420 images for the test set* seems to be a description error in our paper. The accuracy reported in our paper is achieved on the validation set, which consists of 389 test images. You can use the test.py file we provided to test the open-source model files.

Lastly, 

I apologize for not replying to your question promptly. If it is convenient for you, you can contact me via email at [2443434059@qq.com](mailto:2443434059@qq.com).

**Comment by xingchenzhang** [comment]
> Regarding some methods for the experiment accuracy not meeting expectations, I have some ideas.
> 
> 1. Due to the presence of inaccurate annotations in the dataset and a severe class distribution imbalance, the training results are not stable. If the training results can be replicated, I believe an mIoU between 58 and 59 is reasonable.
> 2. In addition, there is a technique related to this work. On line 184 in the model/EAEFNet,
>    during training, use:
>    ` x = self.conv(self.se(self.conv4(x3_2))),`
>    during testing use:
>    `x = self.conv(self.conv4(x3_2))`
>    With this operation, the mAcc  score of the model will increase. However, based on our observations during this period, the mAcc metric does not accurately reflect the overall performance of the work. I personally suggest abandoning its use.
> 3. I recommend using the training dataset provided by RTFNet, with batch_size = 5 and lr = 0.02, to conduct experiments that may achieve the expected results.
> 4. If possible, you can provide me with the environment you used and the testing_results_file.txt file at [2443434059@qq.com](mailto:2443434059@qq.com). Based on your results, I will further analyze and provide more specific solutions!
> 
> Regarding discrepancies between the description in some papers and the code, I have some questions that need clarification.
> 
> 1. Regarding the code part, during the experiment, we used a batch_size of 5, lr of 0.02, and 100 epochs to achieve the accuracy of 58.9 as described in the paper. The batch size of 3 in the code is a tuning version used for GPUs with 12GB VRAM (GTX1080ti, RTX2080ti). If your GPU has 24GB VRAM, you can try using batch_size=5 for the experiment.
> 2. The part about _using 420 images for the test set_ seems to be a description error in our paper. The accuracy reported in our paper is achieved on the validation set, which consists of 389 test images. You can use the test.py file we provided to test the open-source model files.
> 
> Lastly,
> 
> I apologize for not replying to your question promptly. If it is convenient for you, you can contact me via email at [2443434059@qq.com](mailto:2443434059@qq.com).

Hi, 

Thank you very much for your detailed reply!

I will send my test results later to your email. 

One thing I found is that, in your "run_own_pth.py",  you are actually reporting recall instead of precision.

Best,
Xingchen

**Comment by matrixgame2018** [comment]
Yes, follow the setting in the RTFNet(https://github.com/yuxiangsun/RTFNet), in the paper mAcc is recall score

**Comment by xingchenzhang** [comment]
> Yes, follow the setting in the RTFNet(https://github.com/yuxiangsun/RTFNet), in the paper mAcc is recall score

OK. Thanks a lot!
